test/test_optim.py

```diff
@@ -431,6 +431,21 @@ def test_adam(self):
             with self.assertRaisesRegex(ValueError, "Invalid weight_decay value: -1"):
                 optimizer(None, lr=1e-2, weight_decay=-1)
 
+    # Test whether variance parameter is always real
+    def test_complex_adam_variance(self):
+        complex_param = torch.randn(5, 5, dtype=torch.complex64, requires_grad=True)
+        target = torch.randn(5, 5, dtype=torch.complex64)
+        optimizer = optim.Adam([complex_param], lr=0.001)
+
+        for i in range(20):
+            optimizer.zero_grad()
+            loss = (complex_param - target).pow(2).sum()
+            loss.backward()
+            optimizer.step()
+            for idx in optimizer.state_dict()['state'].keys():
+                variance = optimizer.state_dict()['state'][idx]['exp_avg_sq']
+                self.assertEqual(variance.imag, torch.zeros(variance.imag.shape))
```

Comment:
nit - `self.assertEqual(variance.imag, torch.zeros_like(variance.imag))`

Reply:
nice suggestion, unfortunately code was already committed. But i can keep this in mind for the next time.